package/MDAnalysis/coordinates/PDB.py

```diff
@@ -360,20 +326,15 @@ def _read_next_timestep(self, ts=None):
 
     def _read_frame(self, frame):
         try:
-            line = self.frames[frame]
-        except KeyError:
+            offset = self._offsets[frame]
+        except IndexError:  # out of range of known frames
             raise IOError
-        if line is None:
-            # single frame file, we already have the timestep
-            return self.ts
 
-        # TODO: only open file once and leave the file open; then seek back and
-        #       forth; should improve performance substantially
         pos = 0
         occupancy = np.ones(self._n_atoms)
         with util.openany(self.filename, 'rt') as f:
```

Comment:
Yeah we still do. Damn this is an issue for HPC applications. I'll open another issue for this.